pytorch_lightning/trainer/training_loop.py

```diff
@@ -501,6 +501,9 @@ def optimizer_step(self, optimizer, opt_idx, batch_idx, train_step_and_backward_
             using_lbfgs=is_lbfgs,
         )
 
+        if not self.trainer._enable_pl_optimizer and is_overridden("optimizer_step", model_ref):
+            self.on_before_zero_grad(optimizer)
```

Comment:
Maybe add a note here that this happens automatically within the lightning optimizer. Now thinking of it, what about:

```
if self.require_manual_optimizer_hook_call:
    self.on_before_zero_grad(optimizer)
```

Reply:
I removed it. It appeared to me it was more a hack for debugging than anything else.